nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -303,23 +303,31 @@ public void untrashDocuments(List<DocumentModel> docs) {
      */
     protected static final Pattern COLLISION_PATTERN = Pattern.compile("(.*)\\.[0-9]{13,}");
 
+    @Override
+    public boolean isMangledName(String docName) {
+        return TRASHED_PATTERN.matcher(docName).matches();
+    }
+
     @Override
     public String mangleName(DocumentModel doc) {
         return doc.getName() + "._" + System.currentTimeMillis() + "_.trashed";
     }
 
     @Override
     public String unmangleName(DocumentModel doc) {
-        String name = doc.getName();
+        return unmangleName(doc.getCoreSession(), doc.getName(), doc.getParentRef());
```

Comment:
This could be the default implementation in the interface.

Reply:
Done, now default on this calls` unmangleName(Session, parentName, docName)` which by default throws an `UnsupportedOperationException`